config/terraform_cloud_workspace.go

```diff
@@ -43,6 +49,7 @@ func (c *TerraformCloudWorkspaceConfig) Copy() *TerraformCloudWorkspaceConfig {
 	o.ExecutionMode = c.ExecutionMode
 	o.AgentPoolID = c.AgentPoolID
 	o.AgentPoolName = c.AgentPoolName
+	o.TerraformVersion = c.TerraformVersion
```

Comment:
Ah, that's true, thanks for catching! I'll update these and also update the unit test to check for this.

Reply:
In an interesting twist, it looks like we don't need `StringCopy`!  We do use it in all our other copy methods, so I'm going to investigate more as to what's going on.

Reply:
Never mind, I was wrong! For the record since I'm probably going to forget all of this again in the future, I had written a test that looked like:

```
update := "update"
c.ExecutionMode = &update
assert.NotEqual(t, tc.c.ExecutionMode, c.ExecutionMode)
```
And this was working with the current state without `StringCopy`, but that's because I'm updating the reference altogether with the second line.

Anyway, going back to what I originally said I was going to do, and add StringCopy and write a test properly, heh.

Reply:
Thanks for looking into this! I encountered something similar with the State Store when storing the config (I have some stuff I have to swing back and fix 🙈 ). Just in case it's helpful, I ended up using `assert.NotSame()`, not sure if it's useful here though.

Reply:
That would work great here! Let me rework that test, I probably don't need to be modifying the value anymore. I also realized I didn't remove the original incorrect code, which I was keeping around while writing the test, so I'll fix that too. Really got some Friday brain today!

Reply:
Good catch! Don't feel like you have to do this today, Monday is totally ok!

Reply:
Removed the old incorrect copy code and simplified the fully configured case to just check that the pointers aren't the same! I still kept it a separate run because the pointer check isn't needed in the nil and empty case.

Reply:
Makes sense, looks good!